enhancements/bare-metal-fulfillment/README.md

```diff
@@ -0,0 +1,360 @@
+
+1. The tenant uses the Fulfillment CLI to delete a HostPool.
+2. The O-SAC solution fulfills the request and removes the Host resources corresponding to the hosts in the HostPool.
+
+### API Extensions
+
+#### HostPools
+
+A tenant requests bare metal resources by creating a HostPool with a desired specification. Once created, O-SAC
+will repeatedly attempt to fulfill the request until the state of the HostPool matches its specification. Tenants
+can update a HostPool specification, and the same reconciliation process will perform the needed operations to change
+the state of the HostPool.
+
+In the following example, the tenant is requesting a HostPool with two fc430 hosts. Each host will have `network1` attached
+as a native VLAN on one physical interface; and a trunk port with `network2` as a native VLAN
+and `network3` as a tagged VLAN on a second physical interface.
+
+    apiVersion: o-sac.openshift.io/v1alpha1
+    kind: HostPool
+    metadata:
+      name: example
+    spec:
+      hostRequests:
+      - resourceClass: fc430
+        numberOfHosts: 2
+
+      # The networkAttachments section controls how networks are connected to
+      # physical interfaces.
+      networkAttachments:
+      # This configuration will match any available interface.
+      - primary: network1
+      - primary: network2
+        vlans:
+        - network3
+
+If the tenant wishes to specify host properties, they can filter hosts by specifying hostSelectors; for example, this
+HostPool specification will require that hosts be located on rack R2:
+
+    apiVersion: o-sac.openshift.io/v1alpha1
+    kind: HostPool
+    metadata:
+      name: example
+    spec:
+      hostRequests:
+      - resourceClass: fc430
+        numberOfHosts: 2
+	hostSelectors:
+	- rack: R2
+
+      # The networkAttachments section controls how networks are connected to
+      # physical interfaces.
+      networkAttachments:
+      # This configuration will match any available interface.
+      - primary: network1
+      - primary: network2
+        vlans:
+        - network3
+
+If they wish to remove a specific host from their HostPool, they can update their
+HostPool specification to reduce the number of hosts and exclude that host by name;
+this prevents the reconciliation loop from simply re-adding the host later. Note
+that this also allows for a tenant to replace a host by keeping the number of
+hosts constant while also excluding the unwanted host.
+
+    apiVersion: o-sac.openshift.io/v1alpha1
+    kind: HostPool
+    metadata:
+      name: example
+    spec:
+      hostRequests:
+      - resourceClass: fc430
+        numberOfHosts: 1
+	hostSelectors:
+	- rack: R2
+	  exclude:
+	  - HostX
+
+      # The networkAttachments section controls how networks are connected to
+      # physical interfaces.
+      networkAttachments:
+      # This configuration will match any available interface.
+      - primary: network1
+      - primary: network2
+        vlans:
+	- network3
```

Comment:
This is mis-indented. I think there are some tabs-vs-spaces issues happening in this document; you should detab it.